h2o-py/h2o/estimators/estimator_base.py

```diff
@@ -330,6 +330,9 @@ def _resolve_model(self, model_id, model_json):
             # build a useful dict of the params
             for p in m._model_json["parameters"]:
                 m.parms[p["name"]] = p
+            # build a useful dict of the effective params
+            for p in m._model_json["effective_parameters"]:
+                m.effective_parms[p["name"]] = p    
```

Comment:
does this code actually do something? it seems to me that m.effective_parms returns a fresh dict so the effect of this code seems to be thrown away? what am I missing?

Reply:
this puts effective parameters from model._model_json["effective_parametes"] to model.effective_parms with all the attributes they have ('_meta', 'name', 'label', 'help', 'required', 'type', 'value' ...). The only difference is, that you can get specific parameter like xgb1.**effective_parms**['fold_assignment'] instead of xgb1._model_json['effective_parameters'][8] - what is a bit nicer for subsequent usage. Then in model_base.py, by model.effective_params you get dict {nameOfParameter1 : valueOfParameter1, nameOfParameter2 : valueOfParameter2, ...} so you can get value of fold assignment parameter of xgb1 model like xgb1.**effective_params**['fold_assignment']. Did it like this in consistency with model._model_json['parameters'], model.parms and model.actual_params. Do you think that we shouldn't have model.effective_parms and I should extract model.effective_params directly from model_json["effective_parameters"]?

Reply:
oh got it, got confused the field effective_parms and @property def effective_params